File: Programmers/kakao2020/posts_and_beams/SolutionTest.java
package posts_and_beams;

import static org.junit.jupiter.api.Assertions.assertArrayEquals;

import org.junit.jupiter.api.Test;

class SolutionTest {

	Solution s = new Solution();
	
	@Test
	void test1() {
		int n = 5;
		int[][] build_frame = {{1,0,0,1},{1,1,1,1},{2,1,0,1},{2,2,1,1},{5,0,0,1},{5,1,0,1},{4,2,1,1},{3,2,1,1}};
		int[][] actual = s.solution(n, build_frame);
		int[][] expected = {{1,0,0},{1,1,1},{2,1,0},{2,2,1},{3,2,1},{4,2,1},{5,0,0},{5,1,0}};
		assertArrayEquals(expected, actual);
	}
	
	@Test
	void test2() {
		int n = 5;
		int[][] build_frame = {{0,0,0,1},{2,0,0,1},{4,0,0,1},{0,1,1,1},{1,1,1,1},{2,1,1,1},{3,1,1,1},{2,0,0,0},{1,1,1,0},{2,2,0,1}};
		int[][] actual = s.solution(n, build_frame);
		int[][] expected = {{0,0,0},{0,1,1},{1,1,1},{2,1,1},{3,1,1},{4,0,0}};
		assertArrayEquals(expected, actual);
	}

}

File: src/main/java/leetcode/easy/ransom_note/SolutionTest.kt
package leetcode.easy.ransom_note

import org.assertj.core.api.Assertions.assertThat
import org.junit.jupiter.api.Test

class SolutionTest {

    val solution = Solution()

    @Test
    fun test1() {
        val ransomNote = "a"
        val magazine = "b"
        assertThat(solution.canConstruct(ransomNote, magazine)).isFalse()
    }

    @Test
    fun test2() {
        val ransomNote = "aa"
        val magazine = "ab"
        assertThat(solution.canConstruct(ransomNote, magazine)).isFalse()
    }

    @Test
    fun test3() {
        val ransomNote = "aa"
        val magazine = "aab"
        assertThat(solution.canConstruct(ransomNote, magazine)).isTrue()
    }

    @Test
    fun test4() {
        val ransomNote = "aaaa"
        val magazine = "aaaaa"
        assertThat(solution.canConstruct(ransomNote, magazine)).isTrue()
    }

}